xmlrpc.gemspec

```diff
@@ -18,6 +18,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
   spec.required_ruby_version = ">= 2.3"
 
+  spec.add_dependency "logger"
```

Comment:
Why do we need this?
xmlrpc doesn't use `Logger` explicitly.

Reply:
My bad, it's a development dependency, so I moved it to the correct location.

Reply:
It seems that our tests also don't use `Logger` explicitly.

Could you share warnings you got?

Reply:
```
$ ruby -v && bundle exec rake test
ruby 3.4.1 (2024-12-25 revision 48d4efcb85) +PRISM [x86_64-linux]
test/test_webrick_server.rb:9: warning: logger was loaded from the standard library, but will no longer be part of the default gems starting from Ruby 3.5.0.
You can add logger to your Gemfile or gemspec to silence this warning.
[...]
```
But I tried to run the test suite with that line removed and everything works, so I guess it's easier to just remove that require from the tests.